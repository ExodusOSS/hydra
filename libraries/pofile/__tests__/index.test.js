const fs = require('fs')
const path = require('path')
const pofile = require('../src/index')

const data = fs.readFileSync(path.join(__dirname, './fixtures/test.po')).toString()

describe('pofile', () => {
  describe('parse', () => {
    test('should not crash', () => {
      expect(pofile.parse(data)).toBeTruthy()
    })

    test('should successfully parse simple po file', () => {
      const instance = pofile.parse(data)

      expect(instance.entries).toHaveLength(7)
      expect(instance.toString()).toMatchSnapshot()
    })
  })

  describe('create', () => {
    test('should not crash', () => {
      expect(pofile.create()).toBeTruthy()
    })
  })

  describe('api', () => {
    const expectEntryVariants = (method) => {
      expect(method('Enter password')).toBeTruthy()

      // variables
      expect(method('Buy {asset}')).toBeTruthy()
      expect(method('Buy {asset.name}')).toBeTruthy()
      expect(method('Buy {0}')).toBeTruthy()
    }

    test('should add entry successfully', () => {
      const instance = pofile.create()

      instance.addEntry({
        id: 'Welcome to Exodus',
        value: 'Bienvenido a Exodus',
        comments: ['Some comment'],
        flags: ['remote-config'],
      })

      expect(instance.entries).toHaveLength(1)
      expect(instance.toString()).toMatchSnapshot()
    })

    test('should escape entry successfully', () => {
      const instance = pofile.create()

      instance.addEntry({
        id: 'Welcome to "Exodus"',
        value: 'Bienvenido a "Exodus"',
        comments: ['Some comment'],
        flags: ['remote-config'],
      })

      expect(instance.entries).toHaveLength(1)
      expect(instance.toString()).toMatchSnapshot()
    })

    test('should add entry on top of existing file', () => {
      const instance = pofile.parse(data)

      instance.addEntry({
        id: 'Welcome to Exodus',
        value: 'Bienvenido a Exodus',
        comments: ['Some comment'],
      })

      expect(instance.entries).toHaveLength(8)
      expect(instance.toString()).toMatchSnapshot()
    })

    test('should merge duplicated entry', () => {
      const instance = pofile.parse(data)

      instance.addEntry({ id: 'Unlock', references: ['src/ui/index.js '] })

      expect(instance.entries).toHaveLength(7)
      expect(instance.toString()).toMatchSnapshot()
    })

    test('should merge entries with different placeholder variable names', () => {
      const instance = pofile.parse(data)

      instance.addEntry({
        id: '{some.object.nesting} balance',
        value: '{some.object.nesting} balance',
      })

      instance.addEntry({
        id: '{another-var-in-another-context} balance',
        value: '{another-var-in-another-context} balance',
      })

      expect(instance.entries).toHaveLength(8)
      expect(instance.toString()).toMatchSnapshot()
    })

    test('should not add duplicated entry with different value', () => {
      const instance = pofile.parse(data)

      expect(() => instance.addEntry({ id: 'Unlock', value: 'Destrabar' })).toThrow(Error)
    })

    test('should tell if entry exists', () => {
      const instance = pofile.parse(data)

      expect(instance.exists('Unlock')).toBeTruthy()
      expect(instance.exists('Other')).toBeFalsy()

      expectEntryVariants(instance.exists)
    })

    test('should get entry by id', () => {
      const instance = pofile.parse(data)

      expect(instance.getEntry('Unlock')).toBeTruthy()

      expectEntryVariants(instance.getEntry)
    })

    test('should add comment to entry', () => {
      const instance = pofile.parse(data)

      instance.getEntry('Unlock').addComment('other comment')

      expect(instance.toString()).toMatchSnapshot()
    })

    test('should add reference to entry', () => {
      const instance = pofile.parse(data)

      instance.getEntry('Unlock').addReference('src/index.js')

      expect(instance.toString()).toMatchSnapshot()
    })

    test('should add flag to entry', () => {
      const instance = pofile.parse(data)

      instance.getEntry('Unlock').addFlag('remote-config')

      expect(instance.toString()).toMatchSnapshot()
    })
  })
})
