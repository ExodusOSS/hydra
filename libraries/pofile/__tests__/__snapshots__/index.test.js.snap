// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pofile api should add comment to entry 1`] = `
"# other comment
#: src/Component.js:76
msgid "Unlock"
msgstr "Desbloquear"

#: src/Component.js:63
msgid "Enter password"
msgstr "Contraseña"

#, remote-config
msgid "Buy Crypto"
msgstr "Comprar Crypto"

#: src/Component.js:80
msgid "Click on \\"Submit\\" button"
msgstr "Click en el boton \\"Submit\\""

#: src/Component.js:42
msgid "Buy {asset}"
msgstr "Comprar {asset}"

#: src/Component.js:10
msgid "Click on <0>Submit</0>"
msgstr "Click en <0>Submit</0>"

#: src/balance/utils.js:10
msgid "{exodus.account.active.name} balance"
msgstr "{exodus.account.active.name} balance"

#: src/Component.js:100
msgid "Stake {asset.displayTicker}"
msgstr "Stake {asset.displayTicker}"
msgctxt "stake.button"

#: src/Component.js:1
msgid "Stake {asset.displayTicker}"
msgstr "{asset.displayTicker}-Staker""
`;

exports[`pofile api should add entry on top of existing file 1`] = `
"#: src/Component.js:76
msgid "Unlock"
msgstr "Desbloquear"

#: src/Component.js:63
msgid "Enter password"
msgstr "Contraseña"

#, remote-config
msgid "Buy Crypto"
msgstr "Comprar Crypto"

#: src/Component.js:80
msgid "Click on \\"Submit\\" button"
msgstr "Click en el boton \\"Submit\\""

#: src/Component.js:42
msgid "Buy {asset}"
msgstr "Comprar {asset}"

#: src/Component.js:10
msgid "Click on <0>Submit</0>"
msgstr "Click en <0>Submit</0>"

#: src/balance/utils.js:10
msgid "{exodus.account.active.name} balance"
msgstr "{exodus.account.active.name} balance"

#: src/Component.js:100
msgid "Stake {asset.displayTicker}"
msgstr "Stake {asset.displayTicker}"
msgctxt "stake.button"

#: src/Component.js:1
msgid "Stake {asset.displayTicker}"
msgstr "{asset.displayTicker}-Staker"

# Some comment
msgid "Welcome to Exodus"
msgstr "Bienvenido a Exodus""
`;

exports[`pofile api should add entry successfully 1`] = `
"# Some comment
#, remote-config
msgid "Welcome to Exodus"
msgstr "Bienvenido a Exodus""
`;

exports[`pofile api should add flag to entry 1`] = `
"#, remote-config
#: src/Component.js:76
msgid "Unlock"
msgstr "Desbloquear"

#: src/Component.js:63
msgid "Enter password"
msgstr "Contraseña"

#, remote-config
msgid "Buy Crypto"
msgstr "Comprar Crypto"

#: src/Component.js:80
msgid "Click on \\"Submit\\" button"
msgstr "Click en el boton \\"Submit\\""

#: src/Component.js:42
msgid "Buy {asset}"
msgstr "Comprar {asset}"

#: src/Component.js:10
msgid "Click on <0>Submit</0>"
msgstr "Click en <0>Submit</0>"

#: src/balance/utils.js:10
msgid "{exodus.account.active.name} balance"
msgstr "{exodus.account.active.name} balance"

#: src/Component.js:100
msgid "Stake {asset.displayTicker}"
msgstr "Stake {asset.displayTicker}"
msgctxt "stake.button"

#: src/Component.js:1
msgid "Stake {asset.displayTicker}"
msgstr "{asset.displayTicker}-Staker""
`;

exports[`pofile api should add reference to entry 1`] = `
"#: src/Component.js:76
#: src/index.js
msgid "Unlock"
msgstr "Desbloquear"

#: src/Component.js:63
msgid "Enter password"
msgstr "Contraseña"

#, remote-config
msgid "Buy Crypto"
msgstr "Comprar Crypto"

#: src/Component.js:80
msgid "Click on \\"Submit\\" button"
msgstr "Click en el boton \\"Submit\\""

#: src/Component.js:42
msgid "Buy {asset}"
msgstr "Comprar {asset}"

#: src/Component.js:10
msgid "Click on <0>Submit</0>"
msgstr "Click en <0>Submit</0>"

#: src/balance/utils.js:10
msgid "{exodus.account.active.name} balance"
msgstr "{exodus.account.active.name} balance"

#: src/Component.js:100
msgid "Stake {asset.displayTicker}"
msgstr "Stake {asset.displayTicker}"
msgctxt "stake.button"

#: src/Component.js:1
msgid "Stake {asset.displayTicker}"
msgstr "{asset.displayTicker}-Staker""
`;

exports[`pofile api should escape entry successfully 1`] = `
"# Some comment
#, remote-config
msgid "Welcome to \\"Exodus\\""
msgstr "Bienvenido a \\"Exodus\\"""
`;

exports[`pofile api should merge duplicated entry 1`] = `
"#: src/Component.js:76
#: src/ui/index.js
msgid "Unlock"
msgstr "Desbloquear"

#: src/Component.js:63
msgid "Enter password"
msgstr "Contraseña"

#, remote-config
msgid "Buy Crypto"
msgstr "Comprar Crypto"

#: src/Component.js:80
msgid "Click on \\"Submit\\" button"
msgstr "Click en el boton \\"Submit\\""

#: src/Component.js:42
msgid "Buy {asset}"
msgstr "Comprar {asset}"

#: src/Component.js:10
msgid "Click on <0>Submit</0>"
msgstr "Click en <0>Submit</0>"

#: src/balance/utils.js:10
msgid "{exodus.account.active.name} balance"
msgstr "{exodus.account.active.name} balance"

#: src/Component.js:100
msgid "Stake {asset.displayTicker}"
msgstr "Stake {asset.displayTicker}"
msgctxt "stake.button"

#: src/Component.js:1
msgid "Stake {asset.displayTicker}"
msgstr "{asset.displayTicker}-Staker""
`;

exports[`pofile api should merge entries with different placeholder variable names 1`] = `
"#: src/Component.js:76
msgid "Unlock"
msgstr "Desbloquear"

#: src/Component.js:63
msgid "Enter password"
msgstr "Contraseña"

#, remote-config
msgid "Buy Crypto"
msgstr "Comprar Crypto"

#: src/Component.js:80
msgid "Click on \\"Submit\\" button"
msgstr "Click en el boton \\"Submit\\""

#: src/Component.js:42
msgid "Buy {asset}"
msgstr "Comprar {asset}"

#: src/Component.js:10
msgid "Click on <0>Submit</0>"
msgstr "Click en <0>Submit</0>"

#: src/balance/utils.js:10
msgid "{exodus.account.active.name} balance"
msgstr "{exodus.account.active.name} balance"

#: src/Component.js:100
msgid "Stake {asset.displayTicker}"
msgstr "Stake {asset.displayTicker}"
msgctxt "stake.button"

#: src/Component.js:1
msgid "Stake {asset.displayTicker}"
msgstr "{asset.displayTicker}-Staker"

msgid "{another-var-in-another-context} balance"
msgstr "{another-var-in-another-context} balance""
`;

exports[`pofile parse should successfully parse simple po file 1`] = `
"#: src/Component.js:76
msgid "Unlock"
msgstr "Desbloquear"

#: src/Component.js:63
msgid "Enter password"
msgstr "Contraseña"

#, remote-config
msgid "Buy Crypto"
msgstr "Comprar Crypto"

#: src/Component.js:80
msgid "Click on \\"Submit\\" button"
msgstr "Click en el boton \\"Submit\\""

#: src/Component.js:42
msgid "Buy {asset}"
msgstr "Comprar {asset}"

#: src/Component.js:10
msgid "Click on <0>Submit</0>"
msgstr "Click en <0>Submit</0>"

#: src/balance/utils.js:10
msgid "{exodus.account.active.name} balance"
msgstr "{exodus.account.active.name} balance"

#: src/Component.js:100
msgid "Stake {asset.displayTicker}"
msgstr "Stake {asset.displayTicker}"
msgctxt "stake.button"

#: src/Component.js:1
msgid "Stake {asset.displayTicker}"
msgstr "{asset.displayTicker}-Staker""
`;
